### ISSUE TITLE
Edit filename

### ISSUE THREAD

**Issue by kuoirene** [amosproj/amos2024ss05-knowledge-graph-extractor#13]
## User story
1. As a user
2. I want to be able to edit the filename of an uploaded file
3. So that better organization of the stored files can be achieved

## Acceptance criteria
* Each file stored should have an edit button.
* When edit is selected, the user should be able to change the filename.
* When saving, the new name should not already exist, throw an error if the name already exists.
* Once a filename has been edited and saved successfully, the overview of all files should reflect the new file name.

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
